src/setup_payload/QRCodeSetupPayloadGenerator.cpp

```diff
@@ -173,8 +173,13 @@ static CHIP_ERROR generateBitSet(PayloadContents & payload, MutableByteSpan & bi
     VerifyOrReturnError(payload.rendezvousInformation.HasValue(), CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorOnFailure(populateBits(bits.data(), offset, payload.rendezvousInformation.Value().Raw(),
                                       kRendezvousInfoFieldLengthInBits, kTotalPayloadDataSizeInBits));
-    ReturnErrorOnFailure(populateBits(bits.data(), offset, payload.discriminator.GetLongValue(),
-                                      kPayloadDiscriminatorFieldLengthInBits, kTotalPayloadDataSizeInBits));
+
+    // If the payload is valid then discriminator will always be long, but we shouldn't die if we're
+    // encoding a short one due to mAllowInvalidPayload being set. Just use the short value in that case.
```

Comment:
should we in this case validate that mAllowInvalidPayload is set before trying to use the short value?

Please add a description of the use case of when we would like to encode invalid/incomplete data to the comment here. This just says "we shoudn't die" (replace this "shouldn't return error" probably) but it does not explain why. It should explain why.

Reply:
"shouldn't die" means "shouldn't die": `GetLongValue()` involves a VerifyOrDie that the value is long.

Reply:
> should we in this case validate that mAllowInvalidPayload is set before trying to use the short value

If `mAllowInvalidPayload` is not set, then we won't get here with a short discriminator because this is now checked by `isValidQRCodePayload`.

My use case here is to allow maybe-invalid payloads to be serialized, for the purpose of deserializing them again. "AllowInvalidPayload" already describes this pretty well?